### DISCUSSION TITLE
How does the scraper collect its data?

### DISCUSSION THREAD

**Discussion by moosterholt** [davidteather/TikTok-Api#802] [Q&A] [answered]
I've been using this API to collect data for my project and I have a quick question.
For the #sideprofilechallenge, there are 10.500 TikToks with 60.600.000 views total. I've been collecting timestamps to create graphs of when the peak of some specific TikTok trends are, but the script can only collect the dates for up to 1800-ish videos, which shouldn't be problem if those 1800 timestamps can give a somewhat accurate impression of the progression of these trends over time. 
Is there any particular order in which the scraper gathers this data, like does it start with the most recent ones / most popular ones or is it completely random?

**Comment by davidteather** [comment, marked as answer]
It returns the same ordering as [#sideprofilechallenge](https://www.tiktok.com/tag/sideprofilechallenge) on TikTok when the user is not logged in. Unfortunately these pages seem to be pretty static and don't change much.

**Reply by davidteather** [reply]
I'd like to allow for more advanced filtering by like time, but I can't seem to find a page on tiktok.com that supports that while not logged in

**Reply by jaybiz911** [reply]
Hi David, thanks for the awesome tool. How about if logged in? Would be great to be able to parse through the whole 10,500 tiktoks somehow, rather than the first ~2000 (which also seem to be static as you say, unfortunately)

On this note, I thought the verify_FP was for logged in sessions? **EDIT: most definitely not, just tested**

thanks very much

**Reply by moosterholt** [reply]
Alright, thanks! It seems like the videos are unfortunately ordered by the number of likes, and seeing how TikTok's user base has immensely grown the last few years, more recent videos are probably overrepresented in the selection of 1800 vids.
